src/analysis/special_functions/exp_log.lean

```diff
@@ -389,107 +421,149 @@ end
+    tendsto_inv_zero_at_top).comp this
 end
 
-lemma continuous_at_log (hx : 0 < x) : continuous_at log x :=
-continuous_within_at.continuous_at (continuous_on_iff_continuous_restrict.2 continuous_log' _ hx)
-  (mem_nhds_sets (is_open_lt' _) hx)
+lemma continuous_on_log : continuous_on log {0}ᶜ :=
+begin
+  rw [continuous_on_iff_continuous_restrict, set.restrict],
+  conv in (log _) { rw [log_of_ne_zero (show (x : ℝ) ≠ 0, from x.2)] },
+  exact exp_order_iso.symm.continuous.comp (continuous_subtype_mk _ continuous_subtype_coe.norm)
+end
 
-/--
-Three forms of the continuity of `real.log` are provided.
-For the other two forms, see `real.continuous_log'` and `real.continuous_at_log`
--/
-lemma continuous_log {α : Type*} [topological_space α] {f : α → ℝ} (h : ∀a, 0 < f a)
-  (hf : continuous f) : continuous (λa, log (f a)) :=
-show continuous ((log ∘ @subtype.val ℝ (λr, 0 < r)) ∘ λa, ⟨f a, h a⟩),
-  from continuous_log'.comp (continuous_subtype_mk _ hf)
+lemma continuous_log' : continuous (λ x : {x : ℝ // 0 < x}, log x) :=
+continuous_on_iff_continuous_restrict.1 $ continuous_on_log.mono $ λ x hx, ne_of_gt hx
 
-end prove_log_is_continuous
+lemma continuous_at_log (hx : x ≠ 0) : continuous_at log x :=
+(continuous_on_log x hx).continuous_at $ mem_nhds_sets is_open_compl_singleton hx
+
+@[simp] lemma continuous_at_log_iff : continuous_at log x ↔ x ≠ 0 :=
+begin
+  refine ⟨_, continuous_at_log⟩,
+  rintros h rfl,
+  exact not_tendsto_nhds_of_tendsto_at_bot tendsto_log_nhds_within_zero _
+    (h.tendsto.mono_left inf_le_left)
+end
 
 lemma has_deriv_at_log_of_pos (hx : 0 < x) : has_deriv_at log x⁻¹ x :=
 have has_deriv_at log (exp $ log x)⁻¹ x,
-from (has_deriv_at_exp $ log x).of_local_left_inverse (continuous_at_log hx)
-  (ne_of_gt $ exp_pos _) $ eventually.mono (mem_nhds_sets is_open_Ioi hx) @exp_log,
+from (has_deriv_at_exp $ log x).of_local_left_inverse (continuous_at_log hx.ne')
+  (ne_of_gt $ exp_pos _) $ eventually.mono (lt_mem_nhds hx) @exp_log,
 by rwa [exp_log hx] at this
 
 lemma has_deriv_at_log (hx : x ≠ 0) : has_deriv_at log x⁻¹ x :=
 begin
-  by_cases h : 0 < x, { exact has_deriv_at_log_of_pos h },
-  push_neg at h,
-  convert ((has_deriv_at_log_of_pos (neg_pos.mpr (lt_of_le_of_ne h hx)))
-    .comp x (has_deriv_at_id x).neg),
-  { ext y, exact (log_neg_eq_log y).symm },
-  { field_simp [hx] }
+  cases hx.lt_or_lt with hx hx,
+  { convert (has_deriv_at_log_of_pos (neg_pos.mpr hx)).comp x (has_deriv_at_neg x),
+    { ext y, exact (log_neg_eq_log y).symm },
+    { field_simp [hx.ne] } },
+  { exact has_deriv_at_log_of_pos hx }
 end
 
+lemma differentiable_at_log (hx : x ≠ 0) : differentiable_at ℝ log x :=
+(has_deriv_at_log hx).differentiable_at
+
+lemma differentiable_on_log : differentiable_on ℝ log {0}ᶜ :=
+λ x hx, (differentiable_at_log hx).differentiable_within_at
+
+@[simp] lemma differentiable_at_log_iff : differentiable_at ℝ log x ↔ x ≠ 0 :=
+⟨λ h, continuous_at_log_iff.1 h.continuous_at, differentiable_at_log⟩
```

Comment:
Last time I looked, we didn't have the analogue of this and the next lemma in other cases where they would be useful: `x⁻¹`, `tan`, etc.  Note to self: add them!